### PR TITLE
Made channel and role designations pull from backend

### PR DESCRIPTION
**Pull request by jmazouri**
This PR addresses #709 .

This adds two endpoints in `RoleController` and `ChannelController`, for retrieving all the enum names from `DesignatedRoleType` and `DesignatedChannelType`, respectively. This way the enums in the frontend don't need to be updated when the "real" enums in the backend change.

Additionally, this PR includes a new UI for assigning role and channel designations which provides a much smoother user experience on both mobile and desktop platforms, and scales much better with additional role/channel designation types.

### CODE REVIEW

File: Modix/Controllers/RoleController.cs [hunk 1]
```diff
@@ -17,6 +19,12 @@ public RoleController(DiscordSocketClient client, IAuthorizationService modixAut
             RoleService = roleService;
         }
 
+        [HttpGet("types")]
```
Comment (JakenVeina) [inline Modix/Controllers/RoleController.cs]: Nitpick: "RoleDesignationTypes" would be more consistent for the endpoint name.

File: Modix/Controllers/ChannelController.cs [hunk 2]
```diff
@@ -17,6 +20,12 @@ public ChannelController(DiscordSocketClient client, IAuthorizationService modix
             ChannelService = channelService;
         }
 
+        [HttpGet("types")]
```
Comment (JakenVeina) [inline Modix/Controllers/ChannelController.cs]: Nitpick: "ChannelDesignationTypes" would be more consistent for the endpoint name.

Comment (jmazouri) [reply Modix/Controllers/ChannelController.cs]: This should be fine, as the full endpoint ends up being `api/config/channels/types`. `api/config/channels/channeldesignationtypes` seems a bit long.

Comment (JakenVeina) [reply Modix/Controllers/ChannelController.cs]: I would ACTUALLY have it be `api/config/channels/designation-types`, but looking at existing endpoints in this controller, we have such things as `api/config/channels/channeldesignations`.

Comment (jmazouri) [reply Modix/Controllers/ChannelController.cs]: Oh, no, those aren't actually named. For example, the route on the controller itself is `[Route("~/api/config/channels")]`, which means that
```csharp
[HttpGet]
public async Task<IActionResult> ChannelDesignations()
```
will just route to any GET request to `/api/config/channels`

Comment (JakenVeina) [reply Modix/Controllers/ChannelController.cs]: Riiiiiiiiight. I'd forgotten how utterly unintuitive MVC controller routes are.

**Pull request merged**